pkg/apiserver/registry/system/supportbundle/rest.go

```diff
@@ -159,7 +159,9 @@ func (r *supportBundleREST) Create(ctx context.Context, obj runtime.Object, _ re
 				r.cache = b
 			}
 		}()
-		r.clean(ctx, b.Filepath, bundleExpireDuration)
+		if b != nil {
+			r.clean(ctx, b.Filepath, bundleExpireDuration)
+		}
```

Comment:
shouldn't we check the value of `err` after calling `collectAgent` / `collectController` above?

Reply:
thanks, I moved it into the annomous function which wraps to handle error returned from `collectAgent / collectController`
```
               func() {
			r.statusLocker.Lock()
			defer r.statusLocker.Unlock()
			if err != nil {
				klog.Errorf("Error when collecting supportBundle: %v", err)
				r.cache.Status = systemv1beta1.SupportBundleStatusNone
				return
			}
			...
		}()
```